pan3d/explorers/analytics.py

```diff
@@ -0,0 +1,576 @@
+        if args.vtkjs:
+            self.local_rendering = "vtkjs"
+
+        # If no GPU, use local rendering
+        if self.local_rendering is None and not has_gpu():
+            self.local_rendering = "wasm"
+
+        # Process CLI
+        self.ctrl.on_server_ready.add(self._process_cli)
+
+        self.state.nan_colors = [
+            [0, 0, 0, 1],
+            [0.99, 0.99, 0.99, 1],
+            [0.6, 0.6, 0.6, 1],
+            [1, 0, 0, 1],
+            [0, 1, 0, 1],
+            [0, 0, 1, 1],
+            [0.9, 0.9, 0.9, 0],
+        ]
+        self.state.nan_color = 2
+
+        self.ui = None
+        self._setup_vtk()
+        self._build_ui()
+        print("forcing update figure", self.state.color_by)
+        self.plotting.on_change_active_plot()
+
+    # -------------------------------------------------------------------------
+    # VTK Setup
+    # -------------------------------------------------------------------------
+
+    def _setup_vtk(self):
+        self.renderer = vtkRenderer(background=(0.8, 0.8, 0.8))
+        self.interactor = vtkRenderWindowInteractor()
+        self.render_window = vtkRenderWindow(off_screen_rendering=1)
+
+        self.render_window.AddRenderer(self.renderer)
+        self.interactor.SetRenderWindow(self.render_window)
+        self.interactor.GetInteractorStyle().SetCurrentStyleToTrackballCamera()
+
+        self.lut = vtkLookupTable()
+        self.source = vtkXArrayRectilinearSource(input=self.xarray)
+
+        # Need explicit geometry extraction when used with WASM
+        self.geometry = vtkDataSetSurfaceFilter(
+            input_connection=self.source.output_port
+        )
+        self.mapper = vtkPolyDataMapper(
+            input_connection=self.geometry.output_port,
+            lookup_table=self.lut,
+        )
+        self.actor = vtkActor(mapper=self.mapper, visibility=0)
+
+        self.interactor.Initialize()
+
+        axes_actor = vtkAxesActor()
+        self.widget = vtkOrientationMarkerWidget()
+        self.widget.SetOrientationMarker(axes_actor)
+        self.widget.SetInteractor(self.interactor)
+        self.widget.SetViewport(0.85, 0, 1, 0.15)
+        self.widget.EnabledOn()
+        self.widget.InteractiveOff()
+
+    # -------------------------------------------------------------------------
+    # Trame API
+    # -------------------------------------------------------------------------
+
+    def start(self, **kwargs):
+        """Initialize the UI and start the server for XArray Viewer."""
+        self.ui.server.start(**kwargs)
+
+    @property
+    async def ready(self):
+        """Start and wait for the XArray Viewer corroutine to be ready."""
+        await self.ui.ready
+
+    @property
+    def state(self):
+        """Returns the current the trame server state."""
+        return self.server.state
+
+    @property
+    def ctrl(self):
+        """Returns the Controller for the trame server."""
+        return self.server.controller
+
```

Comment:
We may want to add to that list of "trame" methods [those ones](https://github.com/Kitware/pan3d/blob/main/pan3d/viewers/preview.py#L512-L517).